src/js/components/submission/SubmissionPage.jsx

```diff
@@ -1,104 +1,81 @@
-                nextStep={nextStep}
-                errorFromStep={errorFromStep} />),
-            (<CrossFileContentContainer
-                submissionID={submissionID}
-                nextStep={nextStep}
-                errorFromStep={errorFromStep} />),
-            (<GenerateEFContainer
-                submissionID={submissionID}
-                nextStep={nextStep}
-                errorFromStep={errorFromStep} />),
-            (<ReviewDataContainer
-                {...this.props}
-                params={{ submissionID }}
-                setStep={setStep}
-                errorFromStep={errorFromStep} />)
-        ];
-    }
-    // get current component className
-    whichClassName() {
-        return classNames[this.props.step];
-    }
-    // current step component
-    whichComponent() {
-        return this.components()[this.props.step];
-    }
-
     loadingMessage() {
-        if (!this.props.isLoading) return null;
-        return (
+        return this.props.loading ? (
             <ReviewLoading />
-        );
+        ) : null;
     }
 
     errorMessage() {
-        const { errorMessage, isError } = this.props;
-        if (!isError) return null;
-        return (<DABSFABSErrorMessage message={errorMessage} />);
+        const { errorMessage, error } = this.props;
+        return error ? (<DABSFABSErrorMessage message={errorMessage} />) : null;
     }
 
     render() {
         const {
-            isLoading,
-            isError,
-            setStepAndRoute,
-            submissionID
+            loading,
+            error,
+            submissionID,
+            currentStep
         } = this.props;
-        let content = this.whichComponent();
-        if (isLoading) content = this.loadingMessage();
-        if (isError) content = this.errorMessage();
-        const step = this.props.step + 1;
-        const className = this.whichClassName();
+        let content;
+        switch (currentStep) {
+            case 1:
+                content = <ValidateDataContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            case 2:
+                content = <GenerateFilesContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            case 3:
+                content = <CrossFileContentContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            case 4:
+                content = <GenerateEFContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            case 5:
+                content = <ReviewDataContainer submissionID={submissionID} errorFromStep={this.props.errorFromStep} />;
+                break;
+            default:
+                content = null;
+        }
+
+        if (loading) content = this.loadingMessage();
+        if (error) content = this.errorMessage();
```

Comment:
If these are already `if` statements, do we need the functions at all? Can we just do
```if (loading) content = <ReviewLoading />```
for example?